tests/test_database.py

```diff
@@ -4666,22 +4676,170 @@ def test_issue_163_dbc_newlines(self):
+        self.assertTrue(re.search('BA_ "SystemSignalLongSymbol" SG_ 1 {} "{}";'.
+                                  format(SIG_LONG[:32], SIG_LONG),
+                                  content))
+
+        # NO long name attributes for objects with short names
+        self.assertFalse(re.search('BA_ "SystemMessageLongSymbol" BO_ 2 ',
+                                   content))
+        self.assertFalse(re.search('BA_ "SystemNodeLongSymbol" {}'.
+                                   format(NODE_SHORT), content))
+        self.assertFalse(re.search('BA_ "SystemSignalLongSymbol" SG_ 2 ',
+                                   content))
+
+        db_readback = cantools.database.load_string(content, 'dbc')
+
+        self.assertEqual(set([MSG_LONG, MSG_SHORT]),
+                         set([msg.name for msg in db_readback.messages]))
+        self.assertEqual(set([SIG_LONG, SIG_SHORT]),
+                         set([sig.name for msg in db_readback.messages
+                              for sig in msg.signals]))
+        self.assertEqual(set([NODE_LONG, NODE_SHORT]),
+                         set([node.name for node in db_readback.nodes]))
+
+    def test_long_names_from_file_multiple_relations(self):
+        """Test if long names are resolved correctly when message has more
+        than 1 sender.
+
+        """
+
+        filename = 'tests/files/dbc/long_names_multiple_relations.dbc'
+        db = cantools.database.load_file(filename)
+        self.assertEqual(db.get_message_by_frame_id(0).name,
+                         'Msg_Long_Name_56789_123456789_123456789')
+        self.assertEqual(db.get_message_by_frame_id(1).name,
+                         'Msg_Long_Name_56789_123456789_123456789_Copy_1')
+        senders = db.get_message_by_frame_id(6).senders
+        self.assertEqual(len(senders), 2, senders)
+        self.assertTrue('Node_6789_123456789_123456789_123456789' in senders)
+        self.assertTrue('Sender_2_aaaaaaaaaaaaaaaaaaaaaaaAAAAAA' in senders)
+
+    def test_unknown_sender(self):
+        """Test warning if message has a sender not listed in the node list.
+
+        """
+
+        NODE_NAME = 'Node_not_in_list'
+        db = cantools.database.Database()
+        msg = cantools.database.can.message.Message(
+                frame_id=1,
+                name="msg_dummy",
+                length=8,
+                signals=[],
+                senders=[NODE_NAME])
+        db.messages.append(msg)
+        dump = db.as_dbc_string()
+        db_readback = cantools.database.load_string(dump, 'dbc')
+        self.assertEqual(db_readback.messages[0].senders, [NODE_NAME])
+
+    def test_database_version(self):
+        # default value if db created from scratch (map None to ''):
+        db = cantools.database.Database()
+        self.assertIsNone(db.version)
+        self.assertTrue(db.as_dbc_string().startswith('VERSION ""'))
+
+        # write access to version attribute
+        MY_VERSION = "my_version"
```

Comment:
Local variables in lower case only. Applies everywhere.

Reply:
Yes, you're right. I'll correct that.